### PR TITLE
Fix virsh event case failure

### PR DESCRIPTION
**Pull request by chloerh**
On ppc machines, detach interface will cost more time. If doing
attach before detach fully completed, then 'add' will be in front
of 'remove' in the later virsh event output, which will cause failures
of tests. Add a wait_for to wait untill detach completed.

Signed-off-by: haizhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
PASS 1-type_specific.io-github-autotest-libvirt.virsh.event.positive_test.virsh_event.test_events.device-added-removed_event.no_timeout.loop

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py [hunk 1]
```diff
@@ -269,12 +269,25 @@ def trigger_events(dom, events_list=[]):
...
                                                 " net0")
-                    time.sleep(2)
+
+                    def _device_detached(count):
+                        tmp_vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
+                        tmp_ifaces = tmp_vmxml.devices.by_device_tag('interface')
+                        return True if len(tmp_ifaces) == count - 1 else False
+
+                    utils_misc.wait_for(lambda: _device_detached(len(ifaces)), 60, step=1,
+                                        text='Waiting for device to be detached.')
```
Comment (Yingshun) [inline libvirt/tests/src/virsh_cmd/domain/virsh_event.py:290]: virsh.detach_device() has a parameter wait_remove_event, you can set it to True.

Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py:290]: Great! I'll try, if it does not affect event getting.

Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: @Yingshun updated!

**Pull request merged**